internal/store/hazelcast_test.go

```diff
@@ -160,7 +162,22 @@ func TestHazelcastStore_Shutdown(t *testing.T) {
 func createFakeDynamicClient() dynamic.Interface {
 	var subscriptions = test.ReadTestSubscriptions("../../testdata/subscriptions.json")
 	var scheme = runtime.NewScheme()
-	return fake.NewSimpleDynamicClient(scheme, subscriptions[0], subscriptions[1])
+
+	// Create a mapping for list kinds to support reconciliation List() operations
+	listKinds := map[schema.GroupVersionResource]string{
+		{
+			Group:    "mygroup",
+			Version:  "v1",
+			Resource: "myresource",
+		}: "MyResourceList",
+		{
+			Group:    "subscriber.horizon.telekom.de",
+			Version:  "v1",
+			Resource: "subscriptions",
+		}: "SubscriptionList",
+	}
```

Comment:
Having a structure as a key and  a string as a value looks a bit weird. I'll have a more detailed look 👀